### DISCUSSION TITLE
Can we have Milvus to put data into different buckets under minio?

### DISCUSSION THREAD

**Discussion by xiaobingxia-at** [milvus-io/milvus#38116] [Ideas & Feature requests]
Hello, right now Milvus requires to specify one min io bucket name and root path, and all data including "index, insertLogs, deltaLogs" will be persisted into the minio bucket name + root path.

I want to have the data to be organized in this way:
- minio: bucket_for_collection_1/..., bucket_for_collection_2..., bucket_for_collection_3...

The main reason is that we want each collection has a min io encryption key, which only be applied at the bucket level.

Thanks!

**Comment by xiaofan-luan** [comment]
> Hello, right now Milvus requires to specify one min io bucket name and root path, and all data including "index, insertLogs, deltaLogs" will be persisted into the minio bucket name + root path.
> 
> I want to have the data to be organized in this way:
> 
> * minio: bucket_for_collection_1/..., bucket_for_collection_2..., bucket_for_collection_3...
> 
> The main reason is that we want each collection has a min io encryption key, which only be applied at the bucket level.
> 
> Thanks!

this is probably not our design goal for milvus.

We are working on BYOK feature but that is Zilliz only and user can specify why Rootkey for different collections
